libraries/automaton_service/automaton_service.lua

```diff
@@ -0,0 +1,110 @@
+local packet = require('packet')
+local struct = require('struct')
+local server = require('shared.server')
+
+
+local data, ftype = server.new(struct.struct({
+    name = {struct.string(0x10)};
+
+    head_id = {struct.int32},
+    frame_id = {struct.int32},
+    attachments = {struct.struct({
+        id = {struct.uint8},
+        slot = {struct.uint8},
+    })[0x0C]},
+
+    skills = {struct.struct({
+        melee = {struct.uint16},
+        melee_max = {struct.uint16},
+        ranged = {struct.uint16},
+        ranged_max = {struct.uint16},
+        magic = {struct.uint16},
+        magic_max = {struct.uint16},
+    })},
+
+    stats = {struct.struct({
+        str             = {struct.uint16},
+        str_modifier    = {struct.uint16},
+        dex             = {struct.uint16},
+        dex_modifier    = {struct.uint16},
+        vit             = {struct.uint16},
+        vit_modifier    = {struct.uint16},
+        agi             = {struct.uint16},
+        agi_modifier    = {struct.uint16},
+        int             = {struct.uint16},
+        int_modifier    = {struct.uint16},
+        mnd             = {struct.uint16},
+        mnd_modifier    = {struct.uint16},
+        chr             = {struct.uint16},
+        chr_modifier    = {struct.uint16},
+    })},
+
+    available_heads = {struct.bitfield(4)},
+    available_frames= {struct.bitfield(4)},
+    available_attachments = {struct.bitfield(32)},
+}))
+
+packet.incoming:register_init({
+    [{0x044, 0x12}] = function(p)
+        data.name = p.pet_name
+        data.head_id = p.automaton_head
+        data.frame_id = p.automaton_frame
```

Comment:
As I mentioned above, I think it makes sense to add the offset onto these variables already. Like I said in the previous comment, we don't need to adhere to the packet data structure, we can define it in whichever way it is most convenient to us. We can discuss this among the team.